### PR TITLE
Add a Getting Started in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ariard**
Our current protocol dev onboarding is a work-in-progress, having a least an explicit "Getting Started" is really needed at its this phase of the project.

### CODE REVIEW

**Comment by ariard** [comment]
Thanks @moneyball for the fixes, took them directly on my commit with credits.

**Review by TheBlueMatt — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,6 +24,26 @@ requests.
...
+---------------
+
+First and foremost, start small.
+
+This doesn't mean don't be ambitious with the breadth and depth of your contributions but rather
+understand the project context and culture before investing an asymmetric number of hours on
+development compared to your merged work.
+
+Even if you have an extensive open source background or sound software engineering skills, consider
+that comprehension by the other set of regular contributors is as much important than its technical
```
Comment (jkczyz) [inline CONTRIBUTING.md:37]: It's ambiguous as to what "its" is referring to. Should it say _is_?

Comment (jkczyz) [inline CONTRIBUTING.md:37]: s/than/as

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,6 +24,26 @@ requests.
...
+correctness.
+
+It's very welcome to ask for review, either on IRC or LDK Slack. And also for reviewers, it's nice
+to provide timelines when you hope to fulfill the request while bearing in mind for both side that's
```
Comment (jkczyz) [inline CONTRIBUTING.md:41]: s/side/sides

**Review by jkczyz — COMMENTED** [review]
Overall, looks good! Just some minor nits.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ariard) [reply CONTRIBUTING.md:37]: Corrected, added code before comprehension.

**Comment by ariard** [comment]
Thanks for review, updated at 039ffee.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -24,6 +24,26 @@ requests.
...
+---------------
+
+First and foremost, start small.
+
+This doesn't mean don't be ambitious with the breadth and depth of your contributions but rather
+understand the project context and culture before investing an asymmetric number of hours on
+development compared to your merged work.
+
+Even if you have an extensive open source background or sound software engineering skills, consider
+that code comprehension by the other set of regular contributors is as much important as its
```
Comment (jkczyz) [inline CONTRIBUTING.md:37]: Would recommend rephrasing "the other set of regular contributors" simply as "the reviewers". Tightening it up makes the connection between "comprehension" and "technical correctness" more apparent without loss of meaning.

**Review by jkczyz — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (ariard) [reply CONTRIBUTING.md]: Thanks, updated at 1dfb2f9

Comment (jkczyz) [reply CONTRIBUTING.md]: Sorry about nit picking this to death. After reading the new version, I think the following wording would be clearer:

> that the reviewers' comprehension of the code is as much important as technical correctness.

Comment (ariard) [reply CONTRIBUTING.md]: That's a minute update, that's fine :)

**Review by jkczyz — APPROVED** [review]
ACK 7e7635d55903f2fac4e44ee3f2928cb2a6c11112

**Pull request merged**